src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,261 @@
+      const timer = setTimeout(() => {
+        setShouldRender(false);
+      }, 100);
+      return () => clearTimeout(timer);
+    }
+  }, [isOpen]);
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl = config?.wallet?.privacyUrl ?? undefined;
+  const termsOfServiceUrl = config?.wallet?.termsUrl ?? undefined;
+
+  const handleSmartWalletConnector = useCallback(() => {
+    // WalletPreference.SMART_WALLET
+    connect({ connector: connectors[0] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleCoinbaseWalletConnector = useCallback(() => {
+    // WalletPreference.EOA
+    connect({ connector: connectors[1] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleWalletConnectConnector = useCallback(() => {
+    const walletConnectConnector = connectors.find(
+      (c) => c.name === 'WalletConnect',
+    );
+    if (walletConnectConnector) {
+      connect({ connector: walletConnectConnector });
+      onClose();
+    }
+  }, [connect, connectors, onClose]);
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      onClose();
+    }
+  };
+
+  const handleModalClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+  };
+
+  // Remove from DOM completely when not rendered
+  if (!shouldRender) {
+    return null;
+  }
+
+  return (
+    <div
+      className={cn(
+        'fixed inset-0 z-50 flex items-center justify-center',
+        isOpen ? 'animate-fadeIn' : 'animate-fadeOut',
+        'bg-black/70',
+        className,
+      )}
+      onClick={onClose}
+      onKeyDown={handleKeyDown}
+      role="presentation"
+      data-testid="modal-overlay"
+    >
+      <div
+        className={cn(
+          border.default,
+          border.radius,
+          background.default,
+          line.default,
+          'w-[323px] p-6 pb-4',
+          'flex flex-col gap-4',
+          'relative',
+          '-translate-x-1/2 -translate-y-1/2 fixed top-1/2 left-1/2',
+          isOpen ? 'animate-fadeIn' : 'animate-fadeOut',
+        )}
+        onClick={handleModalClick}
+        onKeyDown={handleKeyDown}
+        role="dialog"
+        aria-modal="true"
+      >
+        <button
+          type="button"
+          onClick={onClose}
+          className={cn(
+            'absolute top-4 right-4',
+            'flex items-center justify-center',
+            'h-[12px] w-[12px]',
+          )}
+          aria-label="Close modal"
+        >
+          <div
+            className={cn(
+              'relative h-full w-full transition-colors',
+              '[&>svg>path]:hover:fill-[var(--ock-icon-color-foreground-muted)]',
+            )}
+          >
+            {closeSvg}
+          </div>
+        </button>
+
+        {(appLogo || appName) && (
+          <div className="mt-3 flex w-[275px] flex-col items-center gap-3 self-stretch p-2">
```

Comment:
I don't believe there is a way around them if we want specific styling outside of the tailwind built in styles

Reply:
Yeah that's the syntax I'm just saying it's generally bad practice to hardcode pixel values like this especially with Tailwind.

It's fine for some one-offs but it makes it a lot harder for other devs to reason about/change in the future. Plus, if you think of Tailwind as a design system, it's also nice to keep all of our spacing, widths, and sizes within that system.

Probably this case is different, but if there's spacing/width/height values that we're using frequently, we should add them to the Tailwind config so that it's part of the design system

cc @mindapivessa